src/activity_handlers.cpp

```diff
@@ -2343,23 +2343,26 @@ void activity_handlers::repair_item_finish( player_activity *act, player *p )
 
         // Print message explaining why we stopped
         // But only if we didn't destroy the item (because then it's obvious)
-        const bool destroyed = attempt == repair_item_actor::AS_DESTROYED;
-        if( attempt == repair_item_actor::AS_CANT ||
+        const auto destroyed = attempt == repair_item_actor::AS_DESTROYED;
+        const auto cannot_continue_repair =
+            attempt == repair_item_actor::AS_CANT ||
             destroyed ||
-            !actor->can_repair_target( *p, *fix, !destroyed ) ) {
+            !actor->can_repair_target( *p, *fix_location, !destroyed );
+        if( cannot_continue_repair ) {
             // Cannot continue to repair target, select another target.
+            // **Warning**: as soon as the item is popped back, it is destroyed and can't be used anymore!
             act->targets.pop_back();
         }
 
-        const bool event_happened =
+        const auto event_happened =
             attempt == repair_item_actor::AS_FAILURE ||
             attempt == repair_item_actor::AS_SUCCESS ||
             old_level != p->get_skill_level( actor->used_skill );
 
-        const bool need_input =
+        const auto need_input =
             repeat == REPEAT_ONCE ||
-            ( repeat == REPEAT_EVENT && event_happened ) ||
-            ( repeat == REPEAT_FULL && fix->damage() <= 0 );
+            repeat == REPEAT_EVENT && event_happened ||
+            repeat == REPEAT_FULL && ( cannot_continue_repair || fix_location->damage() <= 0 );
```

Comment:
Do need the parentheses here, since we have all the warnings enabled

Reply:
Thanks ifreund! Fixed in next commit. I dumbly followed my IDE which was saying the parentheses were redundant...

Reply:
They are in this case cause of order of operations, but we trigger warnings cause people often write ```x || y && z``` intending to evaluate ```( x || y ) && z``` under the assumption that `||` and `&&` have the same precedence, while `&&` really has the higher precedence.